### ISSUE TITLE
hasModifier/canSymbolHaveModifier Failure

### ISSUE THREAD

**Issue by david-melendez** [missioncommand/old-mil-sym-java#28]
NOTE: _hasModifier_ calls _canSymbolHaveModifier_

A _canSymbolHaveModifier_ call can fall if the specified symbol only has a single a single modifier in the XML file.  For example, the following code will return false instead of true.

`boolean result = SymbolUtilities.canHaveSymbolModifier("G*T*L-----****X", "W", 1);`

I believe that this is because the SymbolConstantsC.xml contains the following definition of the symbol.

```
 <SYMBOL>
    <SYMBOLID>G*T*L-----****X</SYMBOLID>
    <GEOMETRY>line</GEOMETRY>
    <DRAWCATEGORY>15</DRAWCATEGORY>
    <MAXPOINTS>3</MAXPOINTS>
    <MINPOINTS>3</MINPOINTS>
    <MODIFIERS>W</MODIFIERS>
    <DESCRIPTION>Delay</DESCRIPTION>
    <HIERARCHY>2.X.1.8</HIERARCHY>
    <ALPHAHIERARCHY>TACGRP.TSK.DLY</ALPHAHIERARCHY>
    <PATH>Tactical Graphics/Tasks</PATH>
  </SYMBOL>
```

Note that in the MODIFIERS element, the modifier code exists ("W"). However, the code looks for the code with a period appended (i.e., it is looking for "W.").  This occurs in two places in the code (see =====>).  The code needs to change or the XML needs to change.

```
public static boolean canSymbolHaveModifier(String symbolID, String tgModifier, int symStd)
       {
           String basic = null;
           SymbolDef sd = null;
           boolean returnVal = false;

            try
            {

                basic = SymbolUtilities.getBasicSymbolID(symbolID);
                sd = SymbolDefTable.getInstance().getSymbolDef(basic, symStd);
                if(sd != null)
                {
                    int dc = sd.getDrawCategory();
                    if(tgModifier.equals(ModifiersTG.AM_DISTANCE))
                    {
                        switch(dc)
                        {
                            case SymbolDef.DRAW_CATEGORY_RECTANGULAR_PARAMETERED_AUTOSHAPE:
                            case SymbolDef.DRAW_CATEGORY_SECTOR_PARAMETERED_AUTOSHAPE:
                            case SymbolDef.DRAW_CATEGORY_TWO_POINT_RECT_PARAMETERED_AUTOSHAPE: 
                                returnVal = true;
                                break;
                            case SymbolDef.DRAW_CATEGORY_CIRCULAR_PARAMETERED_AUTOSHAPE:
                            case SymbolDef.DRAW_CATEGORY_CIRCULAR_RANGEFAN_AUTOSHAPE:
                                returnVal = true;
                                break;
                            case SymbolDef.DRAW_CATEGORY_LINE:
   ======>             if(sd.getModifiers().indexOf(tgModifier + ".")>-1)
                                    returnVal = true;
                                break;
                            default:
                                returnVal = false;
                        }
                    }
                    else if(tgModifier.equals(ModifiersTG.AN_AZIMUTH))
                    {
                        switch(dc)
                        {
                            case SymbolDef.DRAW_CATEGORY_RECTANGULAR_PARAMETERED_AUTOSHAPE:
                            case SymbolDef.DRAW_CATEGORY_SECTOR_PARAMETERED_AUTOSHAPE:
                                returnVal = true;
                                break;
                            default:
                                returnVal = false;
                        }
                    }
                    else
                    {
  ======>     sd.getModifiers().indexOf(tgModifier + ".")>-1)
                            returnVal = true;
                    }
                }

                return returnVal;


            }
            catch(Exception exc)
            {
                ErrorLogger.LogException("SymbolUtilties", "canSymbolHaveModifier", exc);
            }
            return returnVal;
       }
```

**Comment by michael-spinelli** [comment]
Fixing the XML.  Had another released planned so will incorporate into that.

**Comment by david-melendez** [comment]
Thank you.

David Melendez

 mailto:david.melendez@nova-tech-solutions.com david.melendez@nova-tech-solutions.com

Software Engineer

Nova Tech Solutions

**Comment by michael-spinelli** [comment]
fixed in v0.1.15, issue#29 will be resolved in next release
